synedrion/src/cggmp21/params.rs

```diff
@@ -104,10 +114,39 @@ impl PaillierParams for PaillierProduction112 {
 }
 
 /// Signing scheme parameters.
-// TODO (#27): this trait can include curve scalar/point types as well,
-// but for now they are hardcoded to `k256`.
-pub trait SchemeParams: Debug + Clone + Send + PartialEq + Eq + Send + Sync + 'static {
-    /// Bits of security the parameters ensure.
+pub trait SchemeParams:
+    'static + Debug + Clone + Send + PartialEq + Eq + Send + Sync + Ord + Copy + Serialize + for<'x> Deserialize<'x>
+where
+    <Self::Curve as CurveArithmetic>::ProjectivePoint: FromEncodedPoint<Self::Curve>,
```

Comment:
Some of the bounds look like they could be removed. The `Copy` ones, for example.

Reply:
I assume you mean `Copy` on `SchemeParams`? Yeah I was on the fence on that, and tried both with and without; in the end I think that the reduced noise all over the code base from having it be `Copy` is worth it.

Reply:
No, the `Copy` ones in the `where` clause. The `Copy` on `SchemeParams` is reasonable, I think clippy will complain if it's not present.

Reply:
Removed, good catch.